### PR TITLE
log client connection id with fetchers and mutations

### PR DESCRIPTION
**Pull request by tharanga**
you can now monitor fetch and mutations of a given client
Issue : #520

### CODE REVIEW

**Comment by dormando** [comment]
Thanks! No need to open a separate issue unless you expect a ton of discussion.

Sad that you had to modify so many different parts of the logger. I tried pretty hard to get a good API for it :(

File: logger.c [hunk 1]
```diff
@@ -653,17 +653,18 @@ static void _logger_log_ext_write(logentry *e, item *it, uint8_t bucket) {
  * TODO: This and below should track and reprint the client fd.
```
Comment (tharanga) [inline logger.c]: @dormando we can remove this TODO if it means this change.

**Comment by tharanga** [comment]
Ack on not opening a separate issue for straightforward fixes.

File: logger.c (hunk 1 above)
Comment (dormando) [reply logger.c]: Yeah think so.

File: logger.c [hunk 2]
```diff
@@ -186,9 +186,9 @@ static int _logger_thread_parse_ise(logentry *e, char *scratch) {
 
     uriencode(le->key, keybuf, le->nkey, KEY_MAX_URI_ENCODED_LENGTH);
     total = snprintf(scratch, LOGGER_PARSE_SCRATCH,
-            "ts=%d.%d gid=%llu type=item_store key=%s status=%s cmd=%s ttl=%u clsid=%u\n",
+            "ts=%d.%d gid=%llu type=item_store key=%s status=%s cmd=%s ttl=%u clsid=%u connid=%d\n",
```
Comment (dormando) [inline logger.c]: s/connid/cfd/ or cid? it's the actual file descriptor; either way connid is a bit long considering 'gid' and 'clsid' are shortened.

**Comment by dormando** [comment]
this is clear enough on a read. I was just too lazy to do it before :) thanks for the tests too.

Remove the comment and shorten connid to cfd and I'll merge it.

**Comment by dormando** [comment]
Released in 1.5.17!

**Pull request closed without merging**